designs/2025-bulk-suppressions-api/README.md

```diff
@@ -0,0 +1,305 @@
+- Automatic Application: By integrating bulk suppression handling directly into the existing `lintText()` and `lintFiles()` methods, users who utilize a suppression file (either at the default location or specified via the new option) will automatically benefit simply by updating their ESLint version.
+- Alignment with CLI: This approach aligns the Node.js API behavior *and configuration options* more closely with the established CLI behavior.
+- Non-Breaking: Since the core change only alters behavior when a suppression file is found (based on the new option or default location), it is considered a non-breaking change for existing API consumers.
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+- No `suppressionsLocation` API Option: The initial consideration was to *not* add the `suppressionsLocation` option to the API, only implementing the default lookup in `cwd`. This was simpler but rejected because it lacked full consistency with the CLI\'s `--suppressions-location` flag, preventing API users from specifying a custom file path. Adding the option provides greater flexibility and closer parity with the CLI.
+- Separate API Method for Applying Suppressions: Introducing new methods like `lintTextWithSuppressions()` was rejected as inconsistent and burdensome for users compared to automatic application within existing methods.
+- Including Suppression File Manipulation Options in `lintText`/`lintFiles`: The CLI includes flags like `--suppress-all`, `--suppress-rule <rule-name>`, and `--prune-suppressions` which generate, update, or prune the suppression file based on lint results. Adding corresponding options to the `lintText` and `lintFiles` API methods was considered. However, this approach was rejected because:
+  - It would introduce file-writing side effects into API methods primarily designed for linting (reading and analyzing code). This could lead to unexpected behavior for API consumers.
+  - It blurs the responsibility of the `lintText`/`lintFiles` methods.
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+- Specific implementation examples for the `LegacyESLint` class.
+- Should functionality equivalent to CLI flags like `--suppress-all` or `--suppress-rule` be supported via separate API methods in the future?
```

Comment:
I don't think so. I think the main use case is people will create and manage a suppressions file using the CLI. The IDE use case is more an extension of that rather than a replacement.

Reply:
@nzakas 
I was relieved because I had the same opinion.
It was an open question to confirm if our understanding was aligned. 

Given this direction, I believe it would be better to slightly modify the RFC's title and content to focus on the scope of considering bulk suppressions when running Lint via the Node.js API.
The reason is that with the current content, it might take time for future discussions referencing this RFC to understand that handling the suppression content itself is out of scope.

How do you think?

Reply:
Sounds good to me. Looks like this was already done?

Reply:
Thank you! I haven't made any changes regarding the scope of the RFC yet, but I will add them shortly.

Reply:
fixed: https://github.com/eslint/rfcs/pull/133/commits/00ba3e9e6c7d9dc232908841b6c056a7c8ce4601